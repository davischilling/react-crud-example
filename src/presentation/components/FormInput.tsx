import { FormControl, Grid, TextField, TextFieldProps } from "@mui/material";

type FormInputProps = {
  fullWidth: boolean;
} & TextFieldProps;

export default function FormInput({ fullWidth, ...rest }: FormInputProps) {
  return (
    <Grid item xs={12}>
      <FormControl fullWidth={fullWidth}>
        <TextField {...rest} />
      </FormControl>
    </Grid>
  );
}
