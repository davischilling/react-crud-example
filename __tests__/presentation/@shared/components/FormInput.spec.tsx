import { render } from '@testing-library/react';
import FormInput from '../../../../src/presentation/@shared/components/FormInput';

describe('FormInput', () => {
  it('should render the FormInput', () => {
    const { asFragment } = render(
      <FormInput fullWidth required name="name" label="Name" value={''} onChange={() => {}} />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
