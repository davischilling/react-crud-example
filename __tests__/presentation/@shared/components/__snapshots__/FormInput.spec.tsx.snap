// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput should render the FormInput 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-13i4rnv-MuiGrid-root"
  >
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
    >
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r0:"
          id=":r0:-label"
        >
          Name
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
          >
             *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r0:"
            name="name"
            required=""
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Name *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
